falcon/api_helpers.py

```diff
@@ -121,9 +116,29 @@ def default_serialize_error(req, exception):
             preferred = 'application/xml'
 
     if preferred is not None:
+        resp.append_header('Vary', 'Accept')
```

Comment:
Please add a test to ensure this header is set when the default serializer is used.